tests/testthat/test-layer.r

```diff
@@ -12,6 +12,26 @@ test_that("unknown params create error", {
   expect_error(geom_point(blah = "red"), "Unknown parameters")
 })
 
+test_that("Unknown params create error with validate_params = TRUE", {
+  expect_error(geom_point(blah = "red", validate_params = TRUE),
+               "Unknown parameters")
+})
+
+test_that("Unknown params create warning with validate_params = FALSE", {
+  expect_warning(geom_point(blah = "red", validate_params = FALSE),
```

Comment:
Hey @faizan-khan-iit I would say that when the user specifies `validate_params=FALSE`, there is no need to warn them. Could you instead add an expectation for no warning?